background/backgrounds.go

```diff
@@ -0,0 +1,45 @@
+package background
+
+import (
+	"notes-service/models"
+	"strconv"
+	"time"
+
+	"github.com/bep/debounce"
+	"go.uber.org/zap"
+)
+
+// TODO : What if the note is deleted before debounced for UpdateKeyword
+func (srv *service) AddProcess(lambdaFct func() error, identifier interface{}) error {
+
+	for index := range srv.processes {
+		if srv.processes[index].Identifier == identifier {
+			// TODO cancel the goroutine by srv.processes.task
+			go srv.processes[index].Debounced(func() { return })
+			srv.processes = remove(srv.processes, index)
+		}
+	}
+
+	// Add a process to the list launch the debounce fct
+	lastIndex := len(srv.processes)
+	newProcess := models.Process{
+		Identifier: identifier,
+		Debounced:  debounce.New(models.TimeToSave * time.Second),
```

Comment:
I think the encapsulation around the background process is a bit weird. It's both encapsulated (the identifier is an interface, there are multiple note "Actions", etc) and at the same time the debounce time is hardcoded, a lot of dependencies specific to notes are injected in the service. Maybe we should make a decision to either fully make it a lib or just accept that this code is application specific.

Reply:
I'm open to conversation about it but I don't see any dependencies to notes in this package

Reply:
For the debounced time I didn't know how to put it any advices ?

Reply:
I prefer to use this package like a lib thats why I put an interface as second argument of addprocess

Reply:
You dont need blockRepo and noteRepo in the service you can just delete it 

models.Process should be in package background 

You should check if identifier is nil and not do the loop if so, this would make it so that it doesnt cancel out others actions in any cases 

The time should be a parameter in the function 

Then it should be a generic package with no dependencies 

Btw @rojas-diego did u get how the identifier system works and why its there ?

Reply:
Fully agree with @edouard-sn and yeah it's very clear for the ID system!